app/models/cart.rb

```diff
@@ -7,18 +7,20 @@ class Cart < ApplicationRecord
 
 
   def add_item(item)
-    cart_item=self.cart_items.find_by(item_id:item.id)
 
-    if cart_item
-      cart_item.update(quantity:cart_item[:quantity]+1)
+    cart_item=self.cart_items.find_or_initialize_by(item_id:item.id,cart_id:self.id)
+    if cart_item.new_record?
+      cart_item.save
     else
-      CartItem.create(item_id:item.id,cart_id:self.id,quantity:1)
+      cart_item.update!(quantity:cart_item[:quantity]+1)
```

Comment:
I still don't get the + 1 logic. 
Please create some method in cart item. And Use that here. So that this logic has some meaning.